### PR TITLE
[stable/acs-engine-autoscaler] Add support for over-provision and idle-threshold

### PR DESCRIPTION
**Pull request by wbuchwalter**
Upgraded image to `1.0.0` and added the new commands that come with this release.
cc @ritazh

### CODE REVIEW

**Comment by wbuchwalter** [comment]
@k8s-ci-robot check CLA

File: stable/acs-engine-autoscaler/README.md [hunk 1]
```diff
@@ -107,11 +107,12 @@ Parameter | Description | Default
...
-`acsdeployment`| [OPTIONAL] The name of the deployment used to deploy the kubernetes cluster initially. If not provided, it is defaulted to `azuredeploy` | None.
-`sleeptime`| [OPTIONAL] The number of seconds to sleep between scaling loops. If not provided, a default of 60 seconds is used. | 60
+`acsdeployment`| [OPTIONAL] The name of the deployment used to deploy the kubernetes cluster initially. | `azuredeploy`.
+`sleeptime`| [OPTIONAL] The number of seconds to sleep between scaling loops. | 60
 `ignorepools`| [OPTIONAL] A list of comma seperated pool names the autoscaler should ignore. | None.
-`spareagents`| [OPTIONAL] Number of agents per pool that should always remain up. If not provided, a default of 1 agent is used. | 1
-
+`spareagents`| [OPTIONAL] Number of agents per pool that should always remain up. | 1
+`idle-threshold`| [OPTIONAL] Maximum duration (in seconds) an agent can stay idle before being deleted. | 1800 (30 minutes)
+`over-provision`| [OPTIONAL] Number of extra agents to create when scaling out. | 0
```
Comment (ritazh) [inline stable/acs-engine-autoscaler/README.md]: I think the values should be `idlethreshold` and `overprovision` as you have provided in the yaml file.

File: stable/acs-engine-autoscaler/values.yaml [hunk 2]
```diff
@@ -23,4 +23,8 @@ acsenginecluster:
   #ignorepools:
   ## Optional parameter denominating number of nodes to spare in a pool after scaling
   #spareagents:
+  ## Optional parameter denominating the maximum duration (in seconds) an agent can stay idle before being deleted
+  #idle-threshold:
+  ## Optional parameter denominating the number of extra agents to create when scaling out
+  #over-provision:
```
Comment (ritazh) [inline stable/acs-engine-autoscaler/values.yaml]: Same here. The values should be `idlethreshold` and `overprovision` as you have provided in the yaml file.

File: stable/acs-engine-autoscaler/templates/deployment.yaml [hunk 3]
```diff
@@ -63,6 +63,14 @@ spec:
             - --spare-agents
             - {{ .Values.acsenginecluster.spareagents }}
             {{- end }}
+            {{- if .Values.acsenginecluster.idlethreshold }}
+            - --idle-threshold
+            - {{ .Values.acsenginecluster.idlethreshold}}
+            {{- end }}
+            {{- if .Values.acsenginecluster.overprovision }}
+            - --over-provision
+            - {{ .Values.acsenginecluster.overprovision}}
```
Comment (ritazh) [inline stable/acs-engine-autoscaler/templates/deployment.yaml]: @wbuchwalter Can you pls take a look at how this value is passed in? I think it's looking for an integer and it's passing in a string?

**Comment by ritazh** [comment]
LGTM 👍

**Comment by wbuchwalter** [comment]
@seanknox Could you launch the tests? Thanks!

**Comment by seanknox** [comment]
@k8s-bot ok to test

**Comment by seanknox** [comment]
@wbuchwalter can you bump the chart version since this is a major image change?

**Comment by wbuchwalter** [comment]
@seanknox Done.

**Comment by wbuchwalter** [comment]
Closed in favor of https://github.com/kubernetes/charts/pull/1408.

**Pull request closed without merging**